File: cpp/2015/day1/main.cpp
#include <fstream>
#include <iostream>


int main() {
    int floor = 0;
    int pos = 0;
    std::ifstream file{"text.txt"};
    char ch;

    while (file.get(ch)) {
        if (ch == '(') floor++;
        else if (ch == ')') floor--;
        pos++;
        if (floor < 0) break;
    }

    file.close();

    std::cout << floor << '\n';
    std::cout << pos << '\n';

}

File: python/2015/day1/main.py
with open("text.txt", "r") as file:
    line = file.readline().strip()
#floor = line.count('(') - line.count(')')

floor = 0
for i, char in enumerate(line, start=1):
    floor += 1 if char == '(' else -1
    if floor < 0:
        break

print(floor)
print(i)